spec/controllers/notes_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe NotesController, type: :controller do
+  describe "GET index" do
+    let(:user) { FactoryBot.create(:user) }
+    let(:note1) { FactoryBot.create(:note, user: user) }
+    let(:note2) { FactoryBot.create(:note, user: user, title: "New note") }
+    let(:note3) { FactoryBot.create(:note, user: user, title: "Ruby is awesome") }
+
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      sign_in user
+    end
+
+    context "when search_key is not present in the query string parameter" do
+      it "should list all notes" do
+        get :index
+        expect(assigns(@user)[:notes]).to contain_exactly(note1, note2, note3)
```

Comment:
Curious, how does `assigns(@user)` work? Can we also just test for `assigns[:notes]`?

Reply:
`assigns` checks the value of the instance variables we set in our controller.

I thought `assign(@user)` is where the map is stored but just found out that `assign(@user)` is not the correct usage. Changed it to `assigns[:notes]`